File: Module.Persist.TPM/Model/History/HistoricalSegment.cs
using Core.History;
using Module.Persist.TPM.Model.TPM;

namespace Module.Persist.TPM.Model.History
{
    [AssociatedWith(typeof(Segment))]
    public class HistoricalSegment : BaseHistoricalEntity<System.Guid>
    {
        public string Name { get; set; }
    }
}

File: Module.Frontend.TPM/Client/model/tpm/segment/HistoricalSegment.js
Ext.define('App.model.tpm.segment.HistoricalSegment', {
    extend: 'Ext.data.Model',
    idProperty: '_Id',
    breezeEntityType: 'Segment',
    fields: [
        { name: '_Id', type: 'string', hidden: true },
        { name: '_ObjectId', hidden: true },
        { name: '_User', type: 'string', isDefault: true },
        { name: '_Role', type: 'string', isDefault: true },
        { name: '_EditDate', type: 'date', isDefault: true },
        { name: '_Operation', type: 'string', isDefault: true },
        { name: 'Name', type: 'string', hidden: false, isDefault: true }
    ],
    proxy: {
        type: 'breeze',
        resourceName: 'HistoricalSegments',
        reader: {
            type: 'json',
            totalProperty: 'inlineCount',
            root: 'results'
        }
    }
});
